python/src/equistore/data/array.py

```diff
@@ -80,24 +78,39 @@ def eqs_array_origin(this, origin):
             origin[0] = array_origin
 
         # use storage.XXX.__class__ to get the right type for all functions
-        self.eqs_array.origin = self.eqs_array.origin.__class__(eqs_array_origin)
+        eqs_array.origin = eqs_array.origin.__class__(eqs_array_origin)
 
-        self.eqs_array.data = self.eqs_array.data.__class__(_eqs_array_data)
+        eqs_array.data = eqs_array.data.__class__(_eqs_array_data)
 
-        self.eqs_array.shape = self.eqs_array.shape.__class__(_eqs_array_shape)
-        self.eqs_array.reshape = self.eqs_array.reshape.__class__(_eqs_array_reshape)
-        self.eqs_array.swap_axes = self.eqs_array.swap_axes.__class__(
-            _eqs_array_swap_axes
-        )
+        eqs_array.shape = eqs_array.shape.__class__(_eqs_array_shape)
+        eqs_array.reshape = eqs_array.reshape.__class__(_eqs_array_reshape)
+        eqs_array.swap_axes = eqs_array.swap_axes.__class__(_eqs_array_swap_axes)
 
-        self.eqs_array.create = self.eqs_array.create.__class__(_eqs_array_create)
-        self.eqs_array.copy = self.eqs_array.copy.__class__(_eqs_array_copy)
-        self.eqs_array.destroy = self.eqs_array.destroy.__class__(_eqs_array_destroy)
+        eqs_array.create = eqs_array.create.__class__(_eqs_array_create)
+        eqs_array.copy = eqs_array.copy.__class__(_eqs_array_copy)
+        eqs_array.destroy = eqs_array.destroy.__class__(_eqs_array_destroy)
 
-        self.eqs_array.move_samples_from = self.eqs_array.move_samples_from.__class__(
+        eqs_array.move_samples_from = eqs_array.move_samples_from.__class__(
             _eqs_array_move_samples_from
         )
 
+        self._eqs_array = eqs_array
```

Comment:
From line 70-97 it looks like a simple change of the order in which attributes are assigned. Before you were assigning the object ``eqs_array_t()`` to the ``ArrayWrapper.eqs_array`` attribute and then modifying it, but now you are creating the object, modifying it, then assigning it. It's quite possible I'm missing some deeper implications here - what is the reason for these changes?

Reply:
no deep meaning here =) Assigning a fully constructed object reduces the rightward drift of the code by removing a bunch of `self.` =)

Reply:
Ok I see thanks!